apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/adapter/AdapterCommon.kt

```diff
@@ -116,6 +116,8 @@ internal fun readFromResponseCodeBlock(
 
   val checkedProperties = mutableSetOf<String>()
 
+  val typeName = context.layout.escapedVariableName(__typename)
```

Comment:
Yeah but the code above prefixes all fields indiscriminately so I thought let's escape everything to avoid special cases but it looks like it may be actually more complex like this 😅. I'll see if I can simplify.